django/library/doi.py

```diff
@@ -0,0 +1,953 @@
+        if DataciteRegistrationLog.is_metadata_stale(codebase):
+            logger.debug(f"Metadata is stale. Updating metadata in DataCite...")
+            ok = datacite_api.update_metadata_for_codebase(codebase)
+            if not ok:
+                logger.error(f"Failed to update metadata for codebase {codebase.pk}")
+        else:
+            logger.debug(f"Metadata for codebase {codebase.pk} is in sync!")
+
+        for j, release in enumerate(codebase.releases.all()):
+            logger.debug(
+                f"Processing release {j+1}/{codebase.releases.count()} release.pk={release.pk}..."
+            )
+            input("Press Enter to continue or CTRL+C to quit...")
+
+            if release.peer_reviewed and release.doi:
+                if DataciteRegistrationLog.is_metadata_stale(release):
+                    logger.debug(f"Metadata is stale. Updating metadata in DataCite...")
+                    ok = datacite_api.update_metadata_for_release(release)
+                    if not ok:
+                        logger.error(
+                            f"Failed to update metadata for release {release.pk}"
+                        )
+                else:
+                    logger.debug(f"Metadata for release {release.pk} is in sync!")
+            else:
+                logger.debug(
+                    f'{"Release has no DOI. " if not release.doi else ""}'
+                    + f'{"Release is not peer reviewed." if not release.peer_reviewed else ""} Skipping.'
+                )
+
+    logger.info(
+        f"Metadata updated for all existing (Codebase & CodebaseRelease) DOIs."
+    )
+    """
+    assert correctness
+    """
+    if not DRY_RUN:
+        print(VERIFICATION_MESSAGE)
+        logger.info(f"Checking that Comses metadata is in sync with DataCite...")
+        invalid_codebases = []
+        invalid_releases = []
+
+        results = datacite_api.threaded_metadata_check(all_codebases_with_dois)
+        for pk, is_meta_valid in results:
+            if not is_meta_valid:
+                invalid_codebases.append(pk)
+
+        if invalid_codebases:
+            logger.error(f"Failure. Metadata not in sync with DataCite for {len(invalid_codebases)} codebases: {invalid_codebases}")
+        else:
+            logger.info(
+                f"Success. Metadata in sync with DataCite for all codebases with DOI.")
+
+        all_releases_with_dois = CodebaseRelease.objects.exclude(doi__isnull=True)
+        results = datacite_api.threaded_metadata_check(all_releases_with_dois)
+        for pk, is_meta_valid in results:
+            if not is_meta_valid:
+                invalid_releases.append(pk)
+
+        if invalid_releases:
+            logger.error(f"Failure. Metadata not in sync with DataCite for {len(invalid_releases)} releases: {invalid_releases}")
+        else:
+            logger.info(
+                f"Success. Metadata in sync with DataCite for all releases with DOI.")
+
+
+"""
+RECURRENT TASKS
+"""
```

Comment:
all of these should be available to be invoked from the management command. I also think the input() waits should be made optional so that we can schedule them as cron jobs for example